From 9daffff83069e266cd1894f0c98a38f98358aeef Mon Sep 17 00:00:00 2001
From: Eran Ben-Avi <benavi@marvell.com>
Date: Thu, 15 Aug 2013 11:20:40 +0300
Subject: [PATCH 0968/1825] tdm: alp: fix tal symbols export

https://github.com/MISL-EBU-System-SW/misl-windriver.git linux-3.4.69-14t2-read
commit a4e983b8411f7b011f6b8cd8ef45e33e01f48c2c

	Adapt TAL symbols exporting to linux
	kernel v3.4

Change-Id: I770604ee13e24df393d86b74354bfe22e7594c7c
Signed-off-by: Eran Ben-Avi <benavi@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/3045
Tested-by: Star_Automation <star@marvell.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 arch/arm/plat-armada/linux_oss/mvOs.h              |    1 +
 .../plat-armada/mv_drivers_lsp/mv_phone/tdm/tal.c  |   10 +++++++++-
 .../plat-armada/mv_drivers_lsp/mv_phone/tdm/tal.h  |   16 ++++++++--------
 3 files changed, 18 insertions(+), 9 deletions(-)

diff --git a/arch/arm/plat-armada/linux_oss/mvOs.h b/arch/arm/plat-armada/linux_oss/mvOs.h
index 69cd6ea..35f9125 100644
--- a/arch/arm/plat-armada/linux_oss/mvOs.h
+++ b/arch/arm/plat-armada/linux_oss/mvOs.h
@@ -55,6 +55,7 @@ disclaimer.
 #include <asm/hardirq.h>
 #include <asm/dma.h>
 #include <asm/io.h>
+#include <linux/module.h>
 
 #include <linux/random.h>
 
diff --git a/arch/arm/plat-armada/mv_drivers_lsp/mv_phone/tdm/tal.c b/arch/arm/plat-armada/mv_drivers_lsp/mv_phone/tdm/tal.c
index de2e45c..124c0ab 100644
--- a/arch/arm/plat-armada/mv_drivers_lsp/mv_phone/tdm/tal.c
+++ b/arch/arm/plat-armada/mv_drivers_lsp/mv_phone/tdm/tal.c
@@ -82,7 +82,7 @@ tal_stat_t tal_init(tal_params_t* tal_params, tal_mmp_ops_t* mmp_ops)
 
 	return TAL_STAT_OK;
 }
-
+EXPORT_SYMBOL(tal_init);
 
 /*---------------------------------------------------------------------------*
  * tal_pcm_tx_completion
@@ -113,6 +113,7 @@ tal_stat_t tal_pcm_start(void)
 	tal_tdm_if_register_ops.tdm_if_ctl_ops.ctl_pcm_start();
 	return TAL_STAT_OK;
 }
+EXPORT_SYMBOL(tal_pcm_start);
 
 /*---------------------------------------------------------------------------*
  * tal_pcm_stop
@@ -123,6 +124,7 @@ tal_stat_t tal_pcm_stop(void)
 	tal_tdm_if_register_ops.tdm_if_ctl_ops.ctl_pcm_stop();
 	return TAL_STAT_OK;
 }
+EXPORT_SYMBOL(tal_pcm_stop);
 
 /*---------------------------------------------------------------------------*
  * tal_exit
@@ -133,6 +135,7 @@ tal_stat_t tal_exit(void)
 	tdm_if_exit();
 	return TAL_STAT_OK;
 }
+EXPORT_SYMBOL(tal_exit);
 
 /*---------------------------------------------------------------------------*
  * tal_stats_get
@@ -147,6 +150,7 @@ tal_stat_t tal_stats_get(tal_stats_t* tal_stats)
 
 	return TAL_STAT_OK;
 }
+EXPORT_SYMBOL(tal_stats_get);
 
 /*---------------------------------------------------------------------------*
  * tal_tdm_clk_config
@@ -156,6 +160,7 @@ void tal_tdm_clk_config(void)
 {
 	mvCtrlTdmClkCtrlConfig();
 }
+EXPORT_SYMBOL(tal_tdm_clk_config);
 
 #if defined(MV_TDM_USE_DCO)
 /*---------------------------------------------------------------------------*
@@ -166,6 +171,7 @@ int tal_tdm_clk_get(void)
 {
 	return mvCtrlTdmClkCtrlGet();
 }
+EXPORT_SYMBOL(tal_tdm_clk_get);
 
 /*---------------------------------------------------------------------------*
  * tal_tdm_clk_set
@@ -175,4 +181,6 @@ void tal_tdm_clk_set(int correction)
 {
 	return mvCtrlTdmClkCtrlSet(correction);
 }
+EXPORT_SYMBOL(tal_tdm_clk_set);
+
 #endif
diff --git a/arch/arm/plat-armada/mv_drivers_lsp/mv_phone/tdm/tal.h b/arch/arm/plat-armada/mv_drivers_lsp/mv_phone/tdm/tal.h
index 5c7005b..e62bb81 100644
--- a/arch/arm/plat-armada/mv_drivers_lsp/mv_phone/tdm/tal.h
+++ b/arch/arm/plat-armada/mv_drivers_lsp/mv_phone/tdm/tal.h
@@ -67,15 +67,15 @@ typedef struct {
 } tal_mmp_ops_t;
 
 /* APIs */
-extern tal_stat_t tal_init(tal_params_t *tal_params, tal_mmp_ops_t *mmp_ops);
-extern tal_stat_t tal_stats_get(tal_stats_t *tal_stats);
-extern tal_stat_t tal_pcm_start(void);
-extern tal_stat_t tal_pcm_stop(void);
-extern tal_stat_t tal_exit(void);
-extern void tal_tdm_clk_config(void);
+tal_stat_t tal_init(tal_params_t *tal_params, tal_mmp_ops_t *mmp_ops);
+tal_stat_t tal_stats_get(tal_stats_t *tal_stats);
+tal_stat_t tal_pcm_start(void);
+tal_stat_t tal_pcm_stop(void);
+tal_stat_t tal_exit(void);
+void tal_tdm_clk_config(void);
 #if defined(MV_TDM_USE_DCO)
-extern int tal_tdm_clk_get(void);
-extern void tal_tdm_clk_set(int correction);
+int tal_tdm_clk_get(void);
+void tal_tdm_clk_set(int correction);
 #endif
 
 #endif /* _TAL_H */
-- 
1.7.5.4

